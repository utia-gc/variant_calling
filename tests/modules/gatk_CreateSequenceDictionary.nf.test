nextflow_process {

    name "Test Process gatk_CreateSequenceDictionary"
    script "modules/gatk_CreateSequenceDictionary.nf"
    process "gatk_CreateSequenceDictionary"

    test("gatk_CreateSequenceDictionary creates .dict file for genome fasta") {
        // instantiate helper objects
        def refs = new RefsEnsembl()

        when {
            params {
                publishDirReports = "${outputDir}/reports"
                publishDirData    = "${outputDir}/data"
            }
            process {
                """
                input[0] = file("${refs.getGenome()}")
                """
            }
        }

        then {
            // test process success
            assert process.success

            // test sequence dictionary file produced
            with(process.out.dict) {
                assert size() == 1
                assert get(0) ==~ /^.*\/genome_I\.dict/
            }
        }

    }

}
