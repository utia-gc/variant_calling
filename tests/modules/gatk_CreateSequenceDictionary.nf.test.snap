{
    "gatk_CreateSequenceDictionary creates .dict file for genome fasta": {
        "content": [
            {
                "0": [
                    "genome_I.dict:md5,212626bf1f545aef6d12b8876c86fff5"
                ],
                "dict": [
                    "genome_I.dict:md5,212626bf1f545aef6d12b8876c86fff5"
                ]
            }
        ],
        "timestamp": "2023-11-07T13:09:16.805040798"
    }
}